core/controller/src/main/scala/whisk/core/entitlement/PackageCollection.scala

```diff
@@ -104,7 +107,14 @@ class PackageCollection(entityStore: EntityStore) extends Collection(Collection.
         } onFailure {
             case t: NoDocumentException =>
                 info(this, s"the package does not exist")
-                promise.success(false)
+                // if owner, reject with not found, otherwise reject fail the promise
+                // to reject with unauthorized (this prevents leaking information about
+                // packages in other namespaces
+                if (isOwner) {
+                    promise.failure(RejectRequest(NotFound))
```

Comment:
Because Future composition generally makes for easier to read code and makes for more deterministic code.

Maybe this clarifies the point: to me `Promise` is to `Future` as `var` to `val` (regarding determinism in the code).